src/hooks/useIsValidExecution.ts

```diff
@@ -11,7 +11,7 @@ import ContractErrorCodes from '@/services/contracts/ContractErrorCodes'
 import { sameAddress } from '@/utils/addresses'
 
 const isContractError = <T extends EthersError>(error: T): error is T & { reason: keyof typeof ContractErrorCodes } => {
-  return Object.keys(ContractErrorCodes).includes(error.reason)
+  return Object.keys(ContractErrorCodes).includes(error.reason!)
```

Comment:
I didn't bother with this as it will be removed with #1285

Reply:
I would suggest something along the lines of:

```ts
  const isEthersError = <T extends Error | EthersError>(error: T): error is EthersError
```

Reply:
Why does the function need to handle `Error` and `EthersError` if it is only used in one place? I think we should rework `EthersError` in general. Ethers v6 will add [proper error types](https://github.com/ethers-io/ethers.js/issues/3565#issuecomment-1332668704) but for the time being I would suggest opting for a more simple solution e.g.

```
const isContractError = (error: EthersError) => {
  if (!error.reason) return false

  return Object.keys(ContractErrorCodes).includes(error.reason)
}
```

Reply:
The reasoning behind the union was because a standard error could be caught. Let's keep your solution and adjust the type as suggested in the future.